cluster/buckets.go

```diff
@@ -544,6 +545,131 @@ func GetDailyAvgBucketUsageFromDB(ctx *Context, date time.Time) ([]*BucketMetric
 	return bucketMetrics, nil
 }
 
+// GetTenantsBucketUsage gets the tenant's bucket's usage in a defined period
+func GetTenantsBucketUsage(ctx *Context, fromDate time.Time, toDate time.Time) ([]*BucketMetric, error) {
+	query := `
+		SELECT year, month,
+		bucket, 
+		avg(daily_avg_usage) / pow(1024.0, 4.0) as monthly_avg_usage
+		FROM (
+			SELECT 
+			year, month, day,
+			bucket,
+			avg(usage) as daily_avg_usage
+			FROM(
+				SELECT 
+					 EXTRACT (YEAR FROM a.last_update) as year,
+					 EXTRACT (MONTH FROM a.last_update) as month,
+					 EXTRACT (DAY FROM a.last_update) as day,
+					 bsize.key as bucket,
+					 bsize.value::FLOAT as usage
+				FROM bucket_metrics as a
+				JOIN json_each_text(a.buckets_sizes) bsize on true
```

Comment:
I think we should normalize bucket_sizes to avoid this expensive operation

Reply:
done by adding two columns `bucket_name` and `bucket_size` also this impacted the periodic function that updates the table bucket_metrics (`buckets.go/getTenantMetrics`) and was updated.